plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenShiftProjectCache.java

```diff
@@ -136,35 +162,45 @@ public void connectionRemoved(IConnection connection) {
 		flushFor(conn);
 	}
 
-	private IProjectAdapter newProjectAdapter(Collection<IProjectAdapter> adapters, IOpenShiftConnection conn, IProject project) {
+	private IProjectAdapter addNewProjectAdapter(Collection<IProjectAdapter> adapters, IOpenShiftConnection conn, IProject project) {
 		OpenShiftProjectUIModel model = new OpenShiftProjectUIModel(conn, project);
-		if(adapters.add(model)) {;
-			return model;
+		synchronized(adapters) {
+			if (adapters.add(model)) {
+				return model;
+			}
+			return null;
 		}
-		return null;
 	}
-	
-	
-	private void notifyAdd(Collection<IProjectAdapter> adapters) {
+
+	private void notifyAdd(IProjectAdapter adapter) {
+		if (adapter == null) {
+			return;
+		}
 		synchronized (listeners) {
 			try {
-				for (IProjectAdapter a : adapters) {
-					listeners.forEach(l->l.handleAddToCache(this, a));
-				}
-			}catch(Exception e) {
+				listeners.forEach(l -> l.handleAddToCache(this, adapter));
```

Comment:
I kept those catches but for sinle updates only so that notifying the other listeners would still be performed even if 1 of them throws.